src/Analyzer.Utilities/JsonPathResolver.cs

```diff
@@ -71,23 +72,55 @@ public IEnumerable<IJsonPathResolver> Resolve(string jsonPath)
         /// in a "resources" property array at the current scope.
         /// </summary>
         /// <param name="resourceType">The type of resource to find.</param>
+        /// <param name="newCurrentPath">The new current path, used for the recursive call.</param>
+        /// <param name="newCurrentScope">The new current scope, used for the recursive call.</param>
         /// <returns>An enumerable of resolvers with a scope of a resource of the specified type.</returns>
-        public IEnumerable<IJsonPathResolver> ResolveResourceType(string resourceType)
+        public IEnumerable<IJsonPathResolver> ResolveResourceType(string resourceType, string newCurrentPath = null, JToken newCurrentScope = null)
         {
-            string fullPath = currentPath + ".resources[*]";
+            var resourceTypeSeparator = "/";
+
+            newCurrentPath ??= this.currentPath;
+            newCurrentScope ??= this.currentScope;
+
+            string fullPath = newCurrentPath + ".resources[*]";
             if (!resolvedPaths.TryGetValue(fullPath, out var resolvedTokens))
             {
-                var resources = this.currentScope.InsensitiveTokens("resources[*]");
+                var resources = newCurrentScope.InsensitiveTokens("resources[*]");
                 resolvedTokens = resources.Select(r => (FieldContent)r).ToList();
                 resolvedPaths[fullPath] = resolvedTokens;
             }
 
+            // When trying to resolve Microsoft.Web/sites/siteextensions, for example, we should consider that siteextensions can be a child resource of Microsoft.Web/sites (a prefix of the original resource type)
+            var resourceTypePrefixes = new List<string> { resourceType };
+            var resourceTypeSuffixes = new List<string> { "" };
+            var indexesOfTypesSeparators = Regex.Matches(resourceType, resourceTypeSeparator).Cast<Match>().Select(m => m.Index).Skip(1);
+            foreach (var indexOfTypeSeparator in indexesOfTypesSeparators)
+            {
+                resourceTypePrefixes.Add(resourceType[..indexOfTypeSeparator]);
+                resourceTypeSuffixes.Add(resourceType[(indexOfTypeSeparator + 1)..]);
+            }
```

Comment:
It appears resourceTypeSuffixes is not significantly used (only in two places):
1. Found resource matches prefix, and suffix is empty (lines 107+109) - this is equivalent to checking if the resource type matches the full `resourceType`.
2. prefix and suffix are concatenated together with '/' to search again recursively (line 116) - this just gets back the original `resourceType` string again.

You should be able to just build a list of 'parent' types and search into those resources, looking for the full original type string.

Reply:
I'm not sure if I understand the suggestion (the last line), you just mean I shouldn't calculate the suffixes and maybe rename the prefixes to "parents"? If that's the case: I agree, the suffixes thing is a leftover from a previous solution, I didn't notice it doesn't make sense anymore

Reply:
Yes, that's what I meant.